jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java

```diff
@@ -153,10 +154,13 @@ public TimerJobInstance getTimerByName(String jobName) {
     				EjbTimerJob job = (EjbTimerJob) info;
     				
     				EjbGlobalJobHandle handle = (EjbGlobalJobHandle) job.getTimerJobInstance().getJobHandle();
-    				localCache.putIfAbsent(jobName, handle.getTimerJobInstance());
-    				if (handle.getUuid().equals(jobName)) {
-    					logger.debug("Job  {} does match timer and is going to be returned", jobName);
+    				
+    				if (handle.getUuid().equals(jobName)) {    					
     					found = handle.getTimerJobInstance();
+    					localCache.putIfAbsent(jobName, found);
```

Comment:
I was wondering about this scenario:
if the cache already contains a key with the same jobName, currently is skipping the insertion (because it's considered to be the same value associated to that key), but if it's not the case -not sure if the scenario is possible-, a warning log may help to detect it.

```
TimerJobInstance tjb = localCache.putIfAbsent(jobName, handle.getTimerJobInstance());
if (!tjb.equals(handle.getTimerJobInstance()))
    logger.warn("Job {} already contained value {} and won't be override with {}", jobName, tjb, handle.getTimerJobInstance());
```

Reply:
this should not be a case as each timer must be uniquely identified

Reply:
thanks, then looks good to me